sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -779,10 +780,16 @@ case object OneRowRelation extends LeafNode {
 }
 
 /** A logical plan for `dropDuplicates`. */
+case object Deduplicate {
+  def apply(keys: Seq[Attribute], child: LogicalPlan): Deduplicate = {
+    Deduplicate(keys, child, child.outputMode)
+  }
+}
+
 case class Deduplicate(
     keys: Seq[Attribute],
     child: LogicalPlan,
-    streaming: Boolean) extends UnaryNode {
+    originalOutputMode: OutputMode) extends UnaryNode {
```

Comment:
The intent here is that callers who need a Deduplicate will use the two-argument form in the Object, which will then use the constructor to preserve the output mode of the child.

A val defined inside the case class isn't accounted for by copy(), which caused test failures when I tried it.